requests/cookies.py

```diff
@@ -144,11 +144,10 @@ def remove_cookie_by_name(cookiejar, name, domain=None, path=None):
     """
     clearables = []
     for cookie in cookiejar:
-        if cookie.name != name:
-            continue
-        if domain is not None and domain != cookie.domain:
-            continue
-        if path is not None and path != cookie.path:
+        bool_cookie = (cookie.name != name)
```

Comment:
> Sandi mentioned to have kind of linear structure when overviewed which is again accomplished by this change.

Sandi talked about heirarchy. She would have been fine with the code as is: there's one level of indentation. Indeed, in Ruby there'd have been none, because you can do each of these as a one-line conditional with a `continue` at the end.

> but wouldn't it be good if multiple if's can be reduced to a single if statement.

No. Unequivocally not. When it comes to code structure, it is rarely the case that one approach is unequivocally better than the other: it depends entirely on the details. For example, here's another way to get all this code into one if statement:

```python
if (name == cookie.name) and (domain is None or domain == cookie.domain) and (path is None or path == cookie.path):
    clearables.append((cookie.domain, cookie.path, cookie.name))
```

But I think we can agree that this conditional is not better than what we're doing right now. If having multiple conditionals is clearer than having one, there's nothing wrong with having multiple conditionals. And in this case, we get a bit of a speedup from it too (avoid doing checks we don't need to do by bailing on them early).